x-pack/filebeat/input/azureeventhub/tracer.go

```diff
@@ -0,0 +1,117 @@
+// to the `logp` package.
+//
+// According to the `github.com/devigned/tab package`,
+// to implement a Tracer, you must provide the following
+// three components:
+//
+// - Tracer
+// - Spanner
+// - Logger
+//
+// Since we are currently only interested in logging, we will
+// implement a Tracer that only logs.
+type logsOnlyTracer struct{}
+
+// ----------------------------------------------------------------------------
+// Tracer
+// ----------------------------------------------------------------------------
+
+// StartSpan returns the input context and a no-op Spanner
+func (nt *logsOnlyTracer) StartSpan(ctx context.Context, operationName string, opts ...interface{}) (context.Context, tab.Spanner) {
+	return ctx, new(logsOnlySpanner)
+}
+
+// StartSpanWithRemoteParent returns the input context and a no-op Spanner
+func (nt *logsOnlyTracer) StartSpanWithRemoteParent(ctx context.Context, operationName string, carrier tab.Carrier, opts ...interface{}) (context.Context, tab.Spanner) {
+	return ctx, new(logsOnlySpanner)
+}
+
+// FromContext returns a no-op Spanner without regard to the input context
+func (nt *logsOnlyTracer) FromContext(ctx context.Context) tab.Spanner {
+	return new(logsOnlySpanner)
+}
+
+// NewContext returns the parent context
+func (nt *logsOnlyTracer) NewContext(parent context.Context, span tab.Spanner) context.Context {
+	return parent
+}
+
+// ----------------------------------------------------------------------------
+// Spanner
+// ----------------------------------------------------------------------------
+
+// logsOnlySpanner is a Spanner implementation that focuses
+// on logging only.
+type logsOnlySpanner struct{}
+
+// AddAttributes is a no-op
+func (ns *logsOnlySpanner) AddAttributes(attributes ...tab.Attribute) {}
+
+// End is a no-op
+func (ns *logsOnlySpanner) End() {}
+
+// Logger returns a Logger implementation
+func (ns *logsOnlySpanner) Logger() tab.Logger {
+	return &logpLogger{logp.L()}
+}
+
+// Inject is no-op
+func (ns *logsOnlySpanner) Inject(carrier tab.Carrier) error {
+	return nil
+}
+
+// InternalSpan returns nil
+func (ns *logsOnlySpanner) InternalSpan() interface{} {
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+// Logger
+// ----------------------------------------------------------------------------
+
+// logpLogger defers logging to the logp package
+type logpLogger struct {
+	logger *logp.Logger
+}
+
+// Info logs a message at info level
+func (sl logpLogger) Info(msg string, attributes ...tab.Attribute) {
+	sl.logger.Info(msg)
+}
+
+// Error logs a message at error level
+func (sl logpLogger) Error(err error, attributes ...tab.Attribute) {
+	sl.logger.Error(err)
+}
+
+// Fatal logs a message at Fatal level
+func (sl logpLogger) Fatal(msg string, attributes ...tab.Attribute) {
+	sl.logger.Fatal(msg)
+}
+
+// Debug logs a message at Debug level
+func (sl logpLogger) Debug(msg string, attributes ...tab.Attribute) {
```

Comment:
[Blocker]

Why are the attributes ignored?
you could use the [`w`](https://github.com/elastic/elastic-agent-libs/blob/main/logp/logger.go#L185-L224) functions to pass the attributes to `logp`

Reply:
With the current simple and log-focused `logsOnlyTracer` and `logsOnlySpanner` implementations, the attributes are `nil`:

![CleanShot 2024-08-13 at 15 04 44@2x](https://github.com/user-attachments/assets/b98d0179-b80a-4c8a-9d10-08a3809c4147)

I am targeting 8.15.1 to ship at least the logs because, as of today, we have literally zero visibility into the legacy SDK's inner workings. The logs would be pure gold in a couple of SDHs I am working on.

Reply:
I aim to bridge the legacy SDK logs into `logp` so they will be available in a diagnostics bundle. 

I would love to have traces and spans. However, logs alone bring 80% of the value, and they seem achievable in the 8.15.1 FF time frame.

Reply:
@AndersonQ, do you think it's okay to ship this logging solution in time for 8.15.1 (this could help in troubleshooting a lot) and invest more in tracing for input v1 and v2 for 8.15.2 and 8.16?

Reply:
well, something is better than nothing, even thought I still believe the effort to add the attributes is marginal, I won't block the PR.